### PR TITLE
Add interfaces for the new logging-db-logic

### PR DESCRIPTION
**Pull request by h8718**
Signed-off-by: Alexander Ertli <alexander.ertli@direktiv.io>

### CODE REVIEW

File: pkg/refactor/logstore/logstore.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	// the timestamp of the log-entry.
+	T     time.Time
+	Msg   string
+	Level Level
+	// Tags contains metadata of the log-entry.
+	Tags util.JSONB
+}
+
+// LogQuery generates a query statement to receive Log-Entries.
+type LogQuery interface {
```
Comment (sir-hassan) [inline pkg/refactor/logstore/logstore.go:30]: It's not clear how the users of LogStore would use this interface as parameter to query logs!

Can you please write an example?

Comment (sir-hassan) [inline pkg/refactor/logstore/logstore.go:30]: It's not clear how the users of LogStore would use this interface as parameter to query logs!

Can you please write an example?

File: pkg/util/jsonb.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package util
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+)
+
+// represents the sql jsonb type.
+type JSONB map[string]interface{}
```
Comment (sir-hassan) [inline pkg/util/jsonb.go:10]: Why sql related types need to be leaked to the public here?

File: pkg/refactor/logstore/logstore.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+const (
+	Server    RecipientType = "server"
+	Namespace RecipientType = "namespace"
+	Workflow  RecipientType = "workflow"
+	Instance  RecipientType = "instance"
+	Mirror    RecipientType = "mirror"
+)
+
+func Convert(recipientType string) (RecipientType, bool) {
```
Comment (sir-hassan) [inline pkg/refactor/logstore/logstore.go:45]: Why users of LogStore need to use this method?

**Review by sir-hassan — CHANGES_REQUESTED** [review]
I added some notes :)

File: pkg/refactor/logstore/logstore.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	T     time.Time
+	Msg   string
+	Level Level
+	// Tags contains metadata of the log-entry.
+	Tags map[string]string
+}
+
+// LogQuery generates a query statement to receive Log-Entries.
+type LogQuery interface {
+	BuildStatement() (string, error)
```
Comment (sir-hassan) [inline pkg/refactor/logstore/logstore.go:33]: It is not clear how this function is used by the user of LogStore?

**Review by sir-hassan — CHANGES_REQUESTED** [review]
I added more comments.

File: pkg/refactor/logstore/logstore.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/google/uuid"
+)
+
+// LogStore manages storing and querying LogEntries.
+type LogStore interface {
+	// appends a log-entry to the logs of the corresponding RecipientID, RecipientType pair.
+	Append(recipientID uuid.UUID, recipientType RecipientType, l LogEntry) error
+	// returns a list of log-entries using the provided LogQuery.
+	// Example usage:
+	// ql := GetLogsQuery(recipienttId, recipientType, loglevel)
```
[the remark this answers is not in the document]
Comment (h8718) [reply pkg/refactor/logstore/logstore.go:16]: Its just an example how the append can be used. I can provide a example implementation of `GetLogsQuery` inside the docs, if you think we need this. 
```
func GetLogsQuery(recipienttId, recipientType, loglevel) {
    return logQ{
      recipienttId: recipienttId,
      recipientType: recipientType,
      loglevel: loglevel,
    }
}

type logQ struct {
      recipienttId recipienttId 
      recipientType recipientType,
      loglevel level,
}

func(lq logQ) BuildStatement() (string,error) {
  if recipientType xy {
      return fmt.Sprinf("SELECT from %s WHERE %s ...", lq.table,lq.recipienttId)
  }
  ...
}

[the remark this answers is not in the document]
Comment (sir-hassan) [reply pkg/refactor/logstore/logstore.go:16]: @h8718 ,

This PR is about to define all the exported interfaces/types needed to make LogStore usable. Please provide all the missing types.

File: pkg/refactor/logstore/logstore.go (hunk 4 above)
Comment (sir-hassan) [reply pkg/refactor/logstore/logstore.go:35]: If its intended to be used inside Get() method, why then its public and exported?

File: pkg/refactor/logstore/logstore.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+// LogQuery generates a query statement to receive Log-Entries.
+type LogQuery interface {
+	// appends a log-entry to the logs of the corresponding RecipientID, RecipientType pair.
+	Append(ctx context.Context, l LogEntry) error
+	// returns a list of log-entries using the provided LogQuery.
+	Get(ctx context.Context, ls LogSelector) ([]*LogEntry, error)
+	// use to build a logSelector for receiving log-entries
+	NewFilterBuilder() LogSelectorBuilder
```
Comment (sir-hassan) [inline pkg/refactor/logstore/logstore.go:39]: NewFilterBuilder() Has nothing to do with the LogQuery interface. This method can be defined in the package directly.

File: pkg/refactor/logstore/logstore.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+// LogSelectorBuilder constructs a LogSelector.
+type LogSelectorBuilder interface {
```
Comment (sir-hassan) [inline pkg/refactor/logstore/logstore.go:43]: I still see that this builder doesn't work, you can verify this by writing a complete usage example.

**Review by sir-hassan — CHANGES_REQUESTED** [review]

**Review by sir-hassan — APPROVED** [review]
Looks nice.

**Pull request merged**